### PR TITLE
file upload handles map and multiple fieldName

### PR DESCRIPTION
**Pull request by snehmehta**
Ticket: https://github.com/EnsembleUI/ensemble/issues/1663

### CODE REVIEW

File: modules/ensemble/lib/action/upload_files_action.dart [hunk 1]
```diff
@@ -145,6 +145,7 @@ Future<void> uploadFiles({
           : (error) =>
               ScreenController().executeAction(context, action.onError!),
       progressCallback: (progress) {
+        if (action.id == null) return;
```
Comment (kmahmood74) [inline modules/ensemble/lib/action/upload_files_action.dart:148]: FileUploadAction.id seems to be required but is defined as optional in FileUploadAction class. Why not just make it required so you don't have to make these checks?

**Review by kmahmood74 — APPROVED** [review]
approved with one comment

**Comment by kmahmood74** [comment]
please remember to update documentation, schema and add a kitchen sink example

File: modules/ensemble/lib/action/upload_files_action.dart (hunk 1 above)
Comment (snehmehta) [reply modules/ensemble/lib/action/upload_files_action.dart:148]: Id is not required. It is used when user want to bind something. Like here if user want to listen to progress of uploading.

I have put check already, in rest of file upload. I saw this was missing throwing exceptions in logs saw just fixed it.

**Pull request merged**